### DISCUSSION TITLE
Proposal to make it possible to create images for another platform

### DISCUSSION THREAD

**Discussion by shawjef3** [testcontainers/testcontainers-java#7315] [Ideas]
It's possible to specify the platform for a container by using `GenericContainer#withCreateContainerCmdModifier`.
However, it's not possible to do the same when creating an image, and so it's
not possible to both create and test an image for a platform.

Add `ImageFromDockerfile#withBuildImageCmdModifier` which will make this possible.

This is implemented by https://github.com/testcontainers/testcontainers-java/pull/7313.

**Comment by shawjef3** [comment]
I see that it's possible to do this by overriding `ImageFromDockerfile#configure`, but that's super not obvious, and requires making a a new class for something that's just a parameter in `GenericContainer`.

**Comment by eddumelendez** [comment]
Don't you think it's a little bit late to open a discussion due to there is a PR already open? :) See our [docs for contributing](https://java.testcontainers.org/contributing/)

**Comment by shawjef3** [comment]
I read the docs for contributing. Can you please link to the PR? I searched them for a couple keywords and it wasn't obvious to me.

EDIT: I'm assuming you aren't talking about my PR since you closed my associated issue.

**Reply by eddumelendez** [reply]
I was talking about yours :)

**Reply by shawjef3** [reply]
Okay. I'll just wait for a code review then. I was trying to retroactively follow the guidelines but I guess that's not required.
